CONTRIBUTING.md

```diff
@@ -133,6 +133,66 @@ We aim to design the core `kale` modules to be highly **reusable**, generic, and
 - Python IDE: [Visual Studio Code](https://code.visualstudio.com/download), [PyCharm](https://www.jetbrains.com/pycharm/download/)
 - GitHub: [GitHub Desktop](https://desktop.github.com/), [GitHub guides](https://guides.github.com/), [GitHub documentations](https://docs.github.com/en)
 
+### Testing
+
+All new code must be covered by unit tests, and regression tests where appropriate. We will extend test coverage to exisiting code. [General advice on testing should be consulted](https://realpython.com/python-testing/), if needed. This will inform you on how to write tests and help with decisions on what aspects of your code need to be tested. There is some subjectivity involved in deciding how much of the potential behaviour of your code to check.
```

Comment:
Many (including me before Googling) may not know what are unit/regression tests. It will be good to add top references for these specific technical terms. The link to realpython is a good reference while it also has lots of ads and limited viewing space, which is not pleasant and efficient (at least to me). 

Will those from the official python docs be good (for beginners)? 

For example, are the following (from the official docs) good and appropriate references?
unit tests: https://docs.python.org/3/library/unittest.html#module-unittest
regression tests: https://docs.python.org/3/library/test.html

Will doctest be useful? https://docs.python.org/3/library/doctest.html

The above references are the official development tools docs: from https://docs.python.org/3/library/development.html

Reply:
https://github.com/ScottishCovidResponse/simple_network_sim/tree/master/tests

Reply:
Add links to official pytest docs.
Add links to examples https://github.com/ScottishCovidResponse/simple_network_sim
Add links to other tutorials etc.

Reply:
"All new code must be covered by unit tests, and regression tests where appropriate.": This would be great of course but it could be hard to achieve in the short term (for my part, at least until I have learned how to write one). We need to see what will be a feasible path and strike a right balance.